### ISSUE TITLE
Conversations disappear after restart

### ISSUE THREAD

**Issue by cknoll** [jabbercat/jabbercat#68]
I joined a MUC, but after restarting JC I have to join it again. It would IMHO be better if active mucs were remembered.

**Comment by sebastianriese** [comment]
This should not happen automatically (you do not want to join all MUCs you ever joined automatically)! However, there should be UI to make a bookmark when joining a MUC and when you are in a MUC, the bookmark then may have the autojoin flag set, which achieves the required behaviour.

Maybe the best UI would be to have to "join" buttons: "join" and "join and bookmark" which reduces the number of clicks required (which may have key bindings of Return and Shift-Return?). Checkboxen are kind of annoying.

**Comment by horazont** [comment]
I intend to add a "star"/"favourite"/"pin" button into the conversation list. This will work for both MUCs and normal contacts.

The idea is to actually always restore the state of conversations on a restart, plus some additional logic:

* Conversations opened on other devices (we will learn that via MAM) will be opened too (unless we also find a ``<gone/>`` Chat State Notification).
* Conversations which have been inactive (not looked at / no message / possibly just XEP-0085 state) for more than a few days will be removed (threshold may be configurable). This happens periodically (but only while the window does not have focus) and on startup. This will emit a ``<gone/>`` Chat State Notification (just like explicitly closing the conversation).
* "Pinned" conversations are never removed.
* "Pinned" MUCs are set to autojoin.

All MUCs ever joined end up in the bookmarks automatically, but without autojoin. They can be removed of course :-) (at least once #67 is fixed :-X).

**Comment by sebastianriese** [comment]
Yes, a "pin" button in the conversation list is indeed a good UI for that! I also like bookmarking all MUCs, it is really UI hell, to have to jump through additional hoops to get a MUC bookmarked!
